DEVELOPING.MD

```diff
@@ -144,6 +144,8 @@ vCenter:
   password: <password>
 ```
 
+If you deploy CPI via the Rancher UI, **do not** `Define vSphere Tags` without specifying the `Region` and `Zone` in which to tag pods. This will cause the provider ID to not be set on the nodes and the deployment to fail. This case should ideally be prevented by UI validation.
```

Comment:
I would remove "This case should ideally be prevented by UI validation.". Are we sure it's UI that should be validating it and not back-end?

Also, curious what is the case of deploying without involving Rancher UI? I assume in that case you can also submit the same not-valid configuration.

```suggestion
If you deploy CPI via the Rancher UI, **do not** `Define vSphere Tags` without specifying the `Region` and `Zone` in which to tag pods. This will cause the provider ID to not be set on the nodes and the deployment to fail.
```

Reply:
@snasovich Yes, it appears so.

I reproduced the issue on the UI and then deployed CPI v1.4.1 into a vsphere cluster via helm command and did not see the issue where the taints are not removed and the provider ID is not set on the nodes. IMO `vSphere Tags` should not be enabled by default.

Test values.yaml
```
vCenter:
  host: <fremont URL>
  port: <port>
  insecureFlag: true
  clusterId: "ablender-test-cluster-id"
  datacenters: <datacenter>
  username: <username>
  password: <password>
  region: ""
  zone: ""
```

Reply:
Hmm, it looks like the region and zone are under `vCenter.labels`: https://github.com/rancher/vsphere-charts/blob/c433c58733e91515157454e46310f8cb58732bea/charts/rancher-vsphere-cpi/values.yaml#L12-L15 

 
Can you give it a second try?

Reply:
Good catch @jiaqiluo. With

```
labels:
  region: ""
  zone: ""
```

Both CPI and CSI charts still deploy successfully and the kubelet removes taints from the nodes on a vSphere rke1 cluster. Still appears to be a ui issue.